src/vecs/adapter/markdown.py

```diff
@@ -0,0 +1,195 @@
+
+                    chunk_word_count = len(current_chunk.split(" "))
+                    line_word_count = len(line.split(" "))
+
+                # back to reasonable code
+                above_max_tokens = (
+                    chunk_word_count + line_word_count > max_tokens
+                    if max_tokens
+                    else False
+                )
+
+            # print(f"{line} is {'not ' if not MarkdownChunker.is_heading(line, next_line) else ''}heading")
+            if MarkdownChunker.is_heading(line, next_line) or above_max_tokens:
+                print(current_chunk[-2:])
+                chunks.append(current_chunk)
+                current_chunk = ""
+
+            current_chunk += f"{line}\n"
+
+        # in case of leftover chunk
+        if current_chunk != "":
+            chunks.append(current_chunk)
+
+        chunks = [
+            chunk[:-1] if chunk.endswith("\n") else chunk for chunk in chunks
+        ]  # remove line breaks added to end of chunk
+
+        return filter(
+            lambda c: not ((c + " ").isspace()), chunks
+        )  # remove empty chunks
+
+    @staticmethod
+    def is_heading(line: str, next_line: str) -> bool:
+        if (line + " ").isspace():
+            return False
+
+        elif line[0] == "#":  # normal markdown headings
+            # we iterate over the string
+            # if we meet a character that is not a '#' without a space preceding it, this is not a valid heading
+            # if we have more than 6 '#' this is also not valid
+
+            last_char = (
+                ""  # initialise to empty string so we don't have a bad comparison
+            )
+            for index, char in enumerate(line):
+
+                if char != "#":
+                    if (
+                        char == " " and last_char == "#"
+                    ):  # end of hashes, start of text, valid heading
+                        return True
+                    elif (
+                        char != " "
+                    ):  # no space between hashes and other characters, invalid heading
+                        return False
+                elif index == 6:  # more than 6 hashes, invalid heading
+                    return False
+                last_char = char
+
+        elif next_line in [
+            "",
+            None,
+        ]:  # if the next line is blank or None then it can't be a two line heading (see below)
+            return False
+
+        elif next_line[0] in [
+            "-",
+            "=",
+        ]:  # markdown also supports headings level 1 and 2 by having a line of equals or dashes underneath respectively
+            return all(
+                char == next_line[0] for char in next_line
+            )  # true if every character in the line is the same (a '-' or a '=')
+
+    def __call__(
+        self,
+        records: Iterable[Tuple[str, Any, Optional[Dict]]],
+        adapter_context: AdapterContext,
+        max_tokens: int = None,
```

Comment:
the type here should be `Optional[int]` or your could change `None` to some arbitrarily large value like `1_000_000` to keep the function signature and its body simpler